### ISSUE TITLE
Bug: dialog crashes on 'open'

### ISSUE THREAD

**Issue by mtrimolet** [spatools/kospa#1]
Calling `dialog.open` throws the following error:
```
index.js:34 Uncaught (in promise) TypeError: Cannot create property 'dialogId' on string 'viewmodels/dialogs/myviewmodel'
    at index.js:34
(anonymous) @ index.js:34
Promise.then (async)
step @ client.js:39
(anonymous) @ client.js:40
__awaiter @ client.js:36
edit @ index.ts:23
(anonymous) @ knockout-delegatedEvents.js:156
```
Example code which generates this error:
```ts
import * as dialog from "@kospa/dialog";
// [...]
return dialog.open({
    viewmodel: "viewmodels/dialogs/myviewmodel",
    view: "views/dialogs/myviewmodel.html",
    template: "views/dialogs/template.html",
    after: after,
    args: args
});
// [...]
```

I think the loader is trying to attribute a value to `viewmodel.dialogId` before the viewmodel is effectively loaded, which might be related to the deferring/promises behavior in these lines:
```ts
// packages/dialog/index.ts:42
export function open<T>(options: DialogOptions): Promise<T> {
    const
        opts = extend({}, defaults, options) as SafeDialogOptions;

    return Promise.resolve(opts.create(opts))
        .then(container => compose(container, opts))
        .then(node => opts.after && opts.after(node, opts))
        .then(() => {
            opts.__dfd = deferred();

            opened[opts.id] = opts;
            (<any>opts.viewmodel).dialogId = opts.id; // <-- The error triggers here

            return opts.__dfd.promise;
        });
}
```

**Comment by mtrimolet** [comment]
It looks like the `bindingComplete` of the TemplateViewModel, which returns a Promise, is not awaited by the `init` function of the bindingHandler `internal_compose` (located in `packages/base/compose.ts`).

**Comment by mtrimolet** [comment]
After further investigations, `bindingComplete` is not supposed to be awaited but `descendantsComplete` is. (2de13060d8f41040d0ccecb56710d0120fa4d7b0)
Therefore child composition in the TemplateViewModel should be handled by the later. (41542caef68f68e84e22f3e613740bc01196c01a)
